tip-xxxx/tiip-did-jp.md

```diff
@@ -0,0 +1,368 @@
+|---|---|---|---|---|---|---|---|
+| data[4] | 0 | 0 | 0 | 0 | 0 | 0 | **q** |
+
+| | ni26 | ni25 | ni24 | ni23 | ni22 | decimal value | encoding char
+|---|---|---|---|---|---|---|---|
+| data[5] | 0 | 0 | 0 | 0 | 0 | 0 | **q** |
+
+| | ni31 | ni30 | ni29 | ni28 | ni27 | decimal value | encoding char
+|---|---|---|---|---|---|---|---|
+| data[6] | 0 | 0 | 0 | 0 | 0 | 0 | **q** |
+
+| | bh03 | bh02 | bh01 | bh00 | vr00 | decimal value | encoding char
+|---|---|---|---|---|---|---|---|
+| data[7] | 0 | 1 | 0 | 1 | 0 | 10 | **2** |
+
+|  | bh08 | bh07 | bh06 | bh05 | bh04 | decimal value | encoding char
+|---|---|---|---|---|---|---|---|
+| data[8] | 0 | 0 | 1 | 0 | 1 | 5 | **9** |
+
+| | bh13 | bh12 | bh11 | bh10 | bh09 | decimal value | encoding char
+|---|---|---|---|---|---|---|---|
+| data[9] | 1 | 1 | 1 | 0 | 0 | 28 | **u** |
+
+| | bh18 | bh17 | bh16 | bh15 | bh14 | decimal value | encoding char
+|---|---|---|---|---|---|---|---|
+| data[10] | 1 | 1 | 0 | 1 | 1 | 27 | **m** |
+
+| | bh23 | bh22 | bh21 | bh20 | bh19 | decimal value | encoding char
+|---|---|---|---|---|---|---|---|
+| data[11] | 0 | 0 | 0 | 0 | 0 | 0 | **q** |
+
+| | ti04 | ti03 | ti02 | ti01 | ti00 | decimal value | encoding char
+|---|---|---|---|---|---|---|---|
+| data[12] | 1 | 0 | 0 | 1 | 0 | 18 | **j** |
+
+| | ti09 | ti08 | ti07 | ti06 | ti05 | decimal value | encoding char
+|---|---|---|---|---|---|---|---|
+| data[13] | 0 | 0 | 1 | 1 | 0 | 6 | **x** |
+
+| | ti14 | ti13 | ti12 | ti11 | ti10 | decimal value | encoding char
+|---|---|---|---|---|---|---|---|
+| data[14] | 0 | 0 | 0 | 0 | 1 | 1 | **p** |
+
+変換後は以下の文字列が得られる。
+
+| | 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12 | 13 | 14 |
+|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|
+| decimal | 8 | 0 | 0 | 0 | 0 | 0 | 0 | 10 | 5 | 28 | 27 | 0 | 18 | 6 | 1 |
+| char | g | q | q | q | q | q | q | 2 | 9 | u | m | q | j | x | p |
+
+**Checksum**
+最後に[BIP-0350](https://github.com/bitcoin/bips/blob/master/bip-0350.mediawiki), Bech32m形式のChecksumを付与する。Checksumとして次の6文字が得られる。
+check_sum=['p', 'k', '2', 'f', 'u', 'x']
+
+最終的なTxRefは次の文字列となる。
+
+```
+gqqqqqq29umqjxppk2fux
+```
+
+**Prefix Exclusion**
+`did:btcr`と同様に`did:tprs`でも[BIP-0173](https://github.com/bitcoin/bips/blob/master/bip-0173.mediawiki)で定義されているPrefixを除外する。
+
+**Output Index=0の時の表現のあいまいさの排除**
+
+`did:btcr`と同様に`did:tprs`もOutput Indexが0の場合は、省略表現のみを用いる。以下に例を挙げる。
+
+* When Network ID=1, Block Height=120739, Transaction Index=10, Output Index=0
+
+```
+Example) No Output Index
+
+VALID:
+gqqqqqqx6t8q2qqllpff-y
+
+INVALID:
+gqqqqqqx6t8q2qqqqqaajauw
+```
+
+* When Network ID=1, Block Height=120739, Transaction Index=10, Output Index=2
+
+```
+例）With Output Index
+
+gqqqqqqx6t8q2qqzqqeeq68h
+```
+
+DID Resolverは無効なformatを検出した場合はDIDを拒否しなければならない。
+
+#### 3.2. TPRS DID Format
+
+`did:tprs`のフォーマットは以下の通り。
+
+```
+tprs-did  = "did:tprs:" tprs-identifier
+             [ ";" did-service ] [ "/" did-path ]
```

Comment:
W3C defines DID syntax, refs: https://w3c.github.io/did-core/#did-syntax
The syntax references [RFC3986](https://www.rfc-editor.org/rfc/rfc3986) and [RFC5234](https://www.rfc-editor.org/rfc/rfc5234).
`;` means a comment. Can we use this for the `did-service`?